src/ssh/cockpitsshoptions.c

```diff
@@ -78,40 +75,43 @@ set_environment_bool (gchar **env,
 }
 
 static gboolean
-get_allow_unknown_hosts (gchar **env)
+get_connect_to_unknown_hosts (gchar **env)
 {
   const gchar *remote_peer = g_environ_getenv (env, "COCKPIT_REMOTE_PEER");
 
+  /* Fallback to allowUnknown  */
+  if (!cockpit_conf_string (COCKPIT_CONF_SSH_SECTION, "connectToUnknownHosts") &&
+      cockpit_conf_bool (COCKPIT_CONF_SSH_SECTION, "allowUnknown", FALSE))
+      return TRUE;
+
   if (g_strcmp0 (remote_peer, "127.0.0.1") == 0 ||
       g_strcmp0 (remote_peer, "::1") == 0 ||
-      cockpit_conf_bool (COCKPIT_CONF_SSH_SECTION, "allowUnknown", FALSE))
-    return TRUE;
+      cockpit_conf_bool (COCKPIT_CONF_SSH_SECTION, "connectToUnknownHosts", FALSE))
+      return TRUE;
 
-  return get_environment_bool (env, "COCKPIT_SSH_ALLOW_UNKNOWN", FALSE);
+  if (!has_environment_val (env, "COCKPIT_SSH_CONNECT_TO_UNKNOWN_HOSTS"))
+      return get_environment_bool (env, "COCKPIT_SSH_ALLOW_UNKNOWN", FALSE);
+  return get_environment_bool (env, "COCKPIT_SSH_CONNECT_TO_UNKNOWN_HOSTS", FALSE);
 }
 
 CockpitSshOptions *
 cockpit_ssh_options_from_env (gchar **env)
 {
 
   CockpitSshOptions *options = g_new0 (CockpitSshOptions, 1);
-  options->knownhosts_data = get_environment_val (env, "COCKPIT_SSH_KNOWN_HOSTS_DATA",
-                                                 NULL);
-  if (g_strcmp0 (options->knownhosts_data, ignore_hosts_data) == 0)
-    options->ignore_hostkey = TRUE;
-
-  if (g_strcmp0 (options->knownhosts_data, authorize_knownhosts_data) == 0)
-    options->knownhosts_authorize = TRUE;
-
   options->knownhosts_file = get_environment_val (env, "COCKPIT_SSH_KNOWN_HOSTS_FILE",
                                                   default_knownhosts);
   options->command = get_environment_val (env, "COCKPIT_SSH_BRIDGE_COMMAND", default_command);
   options->remote_peer = get_environment_val (env, "COCKPIT_REMOTE_PEER", "localhost");
+  options->connect_to_unknown_hosts = get_connect_to_unknown_hosts (env);
 
-  if (options->knownhosts_data != NULL && !options->knownhosts_authorize)
-    options->allow_unknown_hosts = TRUE;
-  else
-    options->allow_unknown_hosts = get_allow_unknown_hosts (env);
+  options->challenge_unknown_host_preconnect =
+      get_environment_bool (env, "COCKPIT_SSH_CHALLENGE_UNKNOWN_HOST_PRECONNECT", FALSE);
+  /* Fallback to _DATA */
```

Comment:
This should get a "HACK:" comment that _DATA is obsolete, and will be removed at some point.

Reply:
Will do